src/Server/Repositories/ClaraJobRepository.cs

```diff
@@ -0,0 +1,236 @@
+            {
+                var inferenceJob = CreateInferenceJob(job, jobName, instances);
+
+                // Makes a copy of the payload to support multiple pipelines per AE Title.
+                // Future, consider use of persisted payloads.
+                MakeACopyOfPayload(inferenceJob);
+
+                await Policy
+                    .Handle<Exception>()
+                    .WaitAndRetryAsync(
+                        3,
+                        retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt)),
+                        (exception, timeSpan, retryCount, context) =>
+                    {
+                        _logger.Log(LogLevel.Error, exception, $"Error saving inference request. Waiting {timeSpan} before next retry. Retry attempt {retryCount}.");
+                    })
+                    .ExecuteAsync(async () =>
+                    {
+                        await _inferenceJobRepository.AddAsync(inferenceJob);
+                        await _inferenceJobRepository.SaveChangesAsync();
+                    })
+                    .ConfigureAwait(false);
+            }
+        }
+
+        public async Task Update(InferenceJob inferenceJob, InferenceJobStatus status)
+        {
+            using var loggerScope = _logger.BeginScope(new LogginDataDictionary<string, object> { { "JobId", inferenceJob.JobId }, { "PayloadId", inferenceJob.PayloadId } });
+            if (status == InferenceJobStatus.Success)
+            {
+                _logger.Log(LogLevel.Information, $"Removing job as completed.");
+                await Delete(inferenceJob);
+            }
+            else
+            {
+                if (++inferenceJob.TryCount > MaxRetryLimit)
+                {
+                    _logger.Log(LogLevel.Information, $"Exceeded maximum job submission retries; removing job from job store.");
+                    await Delete(inferenceJob);
+                }
+                else
+                {
+                    _logger.Log(LogLevel.Debug, $"Adding job back to job store for retry.");
+                    inferenceJob.State = InferenceJobState.Queued;
+                    await UpdateInferenceJob(inferenceJob);
+                }
+            }
+        }
+
+        public async Task<InferenceJob> Take(CancellationToken cancellationToken)
+        {
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                var request = _inferenceJobRepository.FirstOrDefault(p => p.State == InferenceJobState.Queued);
+
+                if (!(request is null))
+                {
+                    using var loggerScope = _logger.BeginScope(new LogginDataDictionary<string, object> { { "JobId", request.JobId }, { "PayloadId", request.PayloadId } });
+                    request.State = InferenceJobState.InProcess;
+                    _logger.Log(LogLevel.Debug, $"Updating request {request.JobId} to InProgress.");
+                    await UpdateInferenceJob(request);
```

Comment:
pass `cancellationToken` along to `UpdateInferenceJob` if possible

Reply:
Fixed 🙏